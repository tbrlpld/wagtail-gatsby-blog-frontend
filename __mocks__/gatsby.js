const React = require('react')
const gatsby = jest.requireActual('gatsby')

module.exports = {
  ...gatsby,
  graphql: jest.fn(),
  Link: jest.fn().mockImplementation(
    // these props are invalid for a default HTML `a` tag.
    ({
      activeClassName,
      activeStyle,
      getProps,
      innerRef,
      partiallyActive,
      ref,
      replace,
      to,
      ...rest
    }) =>
      React.createElement('a', {
        ...rest,
        href: to,
      })
  ),
  StaticQuery: jest.fn(),
  useStaticQuery: jest.fn(),
}
