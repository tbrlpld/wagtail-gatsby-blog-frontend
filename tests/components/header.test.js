
import React from 'react'
import { useStaticQuery } from 'gatsby'

import renderer from 'react-test-renderer'

import Header from '../../src/components/header.js'

describe('Header', () => {
  useStaticQuery.mockImplementationOnce(() => (
    {
      wagtail: {
        pages: [{ 
          id: 1,
          showInMenus: true,
          title: 'Testpage',
          url: 'http://localhost/testpage'
        }]
      },
      site: {
        siteMetadata: {
          title: 'Site Title'
        }
      }
    }
  ))
  it('renders correctly', () => {
    const tree = renderer
      .create(<Header />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
