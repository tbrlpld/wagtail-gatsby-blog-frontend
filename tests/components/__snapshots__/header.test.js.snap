// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header>
  <div
    className="siteTitle"
  >
    <a
      href="/"
    >
      Site Title
    </a>
  </div>
  <nav>
    <span
      style={
        Object {
          "marginRight": "1em",
        }
      }
    >
      <a
        href="http://localhost/testpage"
      >
        Testpage
      </a>
    </span>
  </nav>
</header>
`;
